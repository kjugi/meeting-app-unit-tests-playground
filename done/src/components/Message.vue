<template>
  <div class="message">
    <h2>
      {{ title }}
    </h2>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      required: true
    },
    isMessageShowed: {
      type: Boolean,
      required: true
    }
  },
  watch: {
    isMessageShowed (newStatus, oldStatus) {
      const reference = this

      if (newStatus) {
        setTimeout(function () {
          reference.$emit('hideMessage')
        }, 3000)
      }
    }
  }
}
</script>

<style>
.message {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0 auto;
  padding: 20px;
  border: 1px solid;
  background-color: #fff;
}

.message--success {
  background-color: #3fbc34;
}

.message--error {
  background-color: #ff5353;
}
</style>
