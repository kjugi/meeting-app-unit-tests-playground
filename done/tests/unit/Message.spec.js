import { mount } from '@vue/test-utils'
import Message from '@/components/Message.vue'

// Check for more: https://jestjs.io/docs/en/timer-mocks
jest.useFakeTimers()

const factory = () => {
  return mount(Message, {
    propsData: {
      title: 'test title',
      isMessageShowed: false
    }
  })
}

describe('Message.vue', () => {
  it('renders prop title when passed', () => {
    const wrapper = factory()

    expect(wrapper.find('h2').text()).toMatch('test title')
  })

  it('emit event when isMessageShowed updates', async () => {
    const wrapper = factory()

    expect(wrapper.emitted()).toEqual({})
    expect(wrapper.emitted()).toEqual(expect.not.objectContaining({ hideMessage: [[]] }))

    await wrapper.setProps({ isMessageShowed: true })
    jest.runAllTimers()

    expect(wrapper.emitted('hideMessage')).toBeDefined()
    expect(wrapper.emitted('hideMessage').length).toBe(1)
    expect(wrapper.emitted('hideMessage')[0]).toEqual([])
  })
})
