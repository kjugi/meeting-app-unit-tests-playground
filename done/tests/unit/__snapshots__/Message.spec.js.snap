// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Message.vue default component matches snapshot 1`] = `
<div class="message">
  <h2>
    test title
  </h2>
</div>
`;
